swiftwinrt/code_writers.h

```diff
@@ -1600,41 +1639,33 @@ vtable);
                         param_name,
                         local_param_name);
                 }
-                else if (category == param_category::object_type)
+                else if (needs_wrapper(category))
                 {
-                    w.write("var %: %\n",
-                        local_param_name,
-                        bind<write_type>(*param.type, write_type_params::c_abi));
-
-                    guard.push("% = %\n", get_swift_name(param),
-                        bind<write_consume_type>(param.type, local_param_name));
+                    com_ptr_initialize.push_back(param);
                 }
-                else if (category == param_category::generic_type)
-                {
-                    w.write("var %: %\n",
-                        local_param_name,
-                        bind<write_type>(*param.type, write_type_params::c_abi));
+            }
+        }
 
-                    guard.push("% = %.from(abi: %)\n",
-                        param_name,
-                        bind_bridge_fullname(param.type),
-                        local_param_name);
-                }
+        if (!com_ptr_initialize.empty())
+        {
+            w.write("let (%) = try ComPtrs.initialize { (%) in\n",
+                bind<write_param_names>(com_ptr_initialize, "%"),
+                bind<write_param_names>(com_ptr_initialize, "%Abi"));
+            guard.push_indent();
+            guard.push("}\n");
+
+            for (const auto& param : com_ptr_initialize)
```

Comment:
We have initializers for up to three pointers.  Can this exceed that?  Or am I reading it totally wrong? :)

Reply:
it could get up to 5, but super unlikely any API will ever specify that